doc/developers/contributing.rst

```diff
@@ -619,16 +619,29 @@ Finally, follow the formatting rules below to make it consistently good:
         of the mentioned shapes. The default value is
         `np.ones(shape=(n_samples,))`.
 
+    list_param : list of int of shape (n_classes,)
+
+    typed_ndarray : ndarray of shape (n_samples,), dtype=int
+
+    sample_weight : array-like of shape (n_samples,), default=None
+
 In general have the following in mind:
 
     1. Use Python basic types. (``bool`` instead of ``boolean``)
-    2. Use parenthesis for defining shapes: ``array-like of shape (n_samples,)``
-       or ``array-like of shape (n_samples, n_features)``
+    2. Use parenthesis for defining shapes: ``array-like, shape=(n_samples,)``
+       or ``array-like, shape=(n_samples, n_features)``
     3. For strings with multiple options, use brackets:
        ``input: {'log', 'squared', 'multinomial'}``
     4. 1D or 2D data can be a subset of
        ``{array-like, ndarray, sparse matrix, dataframe}``. Note that ``array-like``
        can also be a ``list``, while ``ndarray`` is explicitly only a ``numpy.ndarray``.
+    5. When specifying the data type of an list``, use ``of`` as a delimiter: 
+       ``list of int``.
+    6. When specifying the dtype of an ndarray, use ``dtype=int`` after
```

Comment:
Nit:
```suggestion
    6. When specifying the dtype of an ndarray, use e.g. ``dtype=np.int32` after
```
as  `int` is not a native numpy dtype and will be cast to `np.int_` which is a long integer. So that makes it a somewhat confusing example.